### PR TITLE
Fix inconsistencies

### PR DESCRIPTION
**Pull request by nada-ezziti**
#### Summary
This PR addresses two separate issues:
1. Updates the **Settings Screen** to align its design with the Figma prototype, including adding the switch button.
2. Standardizes failure handling across methods in `UserRepository` to ensure consistent behavior when a user is not found.

Although these issues are unrelated, both required minimal changes, and given the number of open PRs, I opted to include them in a single PR to reduce overhead.

---

#### **Settings Screen Updates**
- Moved the options section below the profile editing section to match the Figma prototype.
- Added the **switch button** as defined to the Figma design.
- Removed the unused `NotImplementedYet` box.

#### **UserRepository Failure Handling**
- Updated all methods to call `onFailure()` when a user is not found.
- Updated corresponding tests to reflect the new behavior.
- Ensured uniform failure handling across all methods in `UserRepository`.

---

Capture d’écran 2024-12-11 à 17 57 57

Let me know if any adjustments are needed!

### CODE REVIEW

File: app/src/main/java/com/github/se/signify/model/common/user/FirestoreUserRepository.kt [hunk 1]
```diff
@@ -75,7 +76,8 @@ class FirestoreUserRepository(
             (documentSnapshot[friendRequestsListPath] as? List<*>)?.filterIsInstance<String>()
         onSuccess(friendsRequests ?: emptyList())
       } else {
-        onSuccess(emptyList())
+        // Call onFailure when user's document is not found
+        onFailure(NoSuchElementException("User not found for ID: $userId"))
```
Comment (samymessouak) [inline app/src/main/java/com/github/se/signify/model/common/user/FirestoreUserRepository.kt:80]: Such a better way to handle this exception, well done

File: app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt [hunk 2]
```diff
@@ -187,11 +171,26 @@ fun SettingsScreen(
...
+          Text(
+              text = modeText,
+              style = MaterialTheme.typography.bodyLarge,
+              color = MaterialTheme.colorScheme.onBackground)
+          Switch(checked = isDarkTheme, onCheckedChange = { onThemeChange(it) })
+        }
+    // Switch between English and French
+    LanguageSwitch(isFrench, onLanguageChange)
+
+    Spacer(modifier = Modifier.height(64.dp))
```
Comment (samymessouak) [inline app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt:192]: If I remember well , we discussed to move those switch under the username, i prefer it like this!

File: app/src/test/java/com/github/se/signify/model/common/user/FirestoreUserRepositoryTest.kt [hunk 3]
```diff
@@ -982,4 +987,184 @@ class FirestoreUserRepositoryTest {
     // Verifies that the `update` operation was called with the expected data
     verify(mockCurrentUserDocRef).update(updatedDataCaptor.firstValue)
   }
+
+  @Suppress("UNCHECKED_CAST")
+  @Test
+  fun getFriendsList_shouldHandleUserDocNotExist() {
```
Comment (samymessouak) [inline app/src/test/java/com/github/se/signify/model/common/user/FirestoreUserRepositoryTest.kt:993]: We did not test if it handle when the user doc does not exist before, its a good thing to add those edge cases for every methods

**Review by samymessouak — APPROVED** [review]
Few changes but it makes the difference ! Well done
